app/src/androidTest/java/org/mozilla/fenix/ui/SponsoredShortcutsTest.kt

```diff
@@ -0,0 +1,298 @@
+            }
+        }.openThreeDotMenu {
+        }.openCustomizeHome {
+            verifySponsoredShortcutsCheckBox(true)
+            clickSponsoredShortcuts()
+            verifySponsoredShortcutsCheckBox(false)
+        }.goBack {
+            verifyNotExistingSponsoredTopSitesList()
+        }
+    }
+
+    @Test
+    fun openSponsoredShortcutTest() {
+        var sponsoredShortcutTitle = ""
+
+        homeScreen {
+            sponsoredShortcutTitle = getSponsoredShortcutTitle(2)
+        }.openSponsoredShortcut(sponsoredShortcutTitle) {
+            verifyUrl(sponsoredShortcutTitle)
+        }
+    }
+
+    @Test
+    fun openSponsoredShortcutInPrivateBrowsingTest() {
+        var sponsoredShortcutTitle = ""
+
+        homeScreen {
+            sponsoredShortcutTitle = getSponsoredShortcutTitle(2)
+        }.openContextMenuOnSponsoredShortcut(sponsoredShortcutTitle) {
+        }.openTopSiteInPrivateTab {
+            verifyUrl(sponsoredShortcutTitle)
+        }
+    }
+
+    @Ignore("Failing, see: https://github.com/mozilla-mobile/fenix/issues/25926")
+    @Test
+    fun verifySponsoredShortcutsSponsorsAndPrivacyOptionTest() {
+        var sponsoredShortcutTitle = ""
+
+        homeScreen {
+            sponsoredShortcutTitle = getSponsoredShortcutTitle(2)
+        }.openContextMenuOnSponsoredShortcut(sponsoredShortcutTitle) {
+        }.clickSponsorsAndPrivacyButton {
+            verifyUrl("support.mozilla.org/en-US/kb/sponsor-privacy")
+        }
+    }
+
+    @Test
+    fun verifySponsoredShortcutsSettingsOptionTest() {
+        var sponsoredShortcutTitle = ""
+
+        homeScreen {
+            sponsoredShortcutTitle = getSponsoredShortcutTitle(2)
+        }.openContextMenuOnSponsoredShortcut(sponsoredShortcutTitle) {
+        }.clickSponsoredShortcutsSettingsButton {
+            verifyHomePageView()
+        }
+    }
+
+    @Test
+    fun verifySponsoredShortcutsDetailsTest() {
+        homeScreen {
+            var sponsoredShortcutTitle = getSponsoredShortcutTitle(2)
+            var sponsoredShortcutTitle2 = getSponsoredShortcutTitle(3)
+
+            verifySponsoredShortcutDetails(sponsoredShortcutTitle, 2)
+            verifySponsoredShortcutDetails(sponsoredShortcutTitle2, 3)
+        }
+    }
+
+    // The default search engine should not be displayed as a sponsored shortcut
+    @Test
+    fun defaultSearchEngineIsNotDisplayedAsSponsoredShortcutTest() {
+        val sponsoredShortcutTitle = "Amazon"
+
+        homeScreen {
+        }.openThreeDotMenu {
+        }.openSettings {
+        }.openSearchSubMenu {
+            changeDefaultSearchEngine(defaultSearchEngine)
+        }
+
+        exitMenu()
+
+        homeScreen {
+            verifySponsoredShortcutDoesNotExist(sponsoredShortcutTitle, 2)
+            verifySponsoredShortcutDoesNotExist(sponsoredShortcutTitle, 3)
+        }
+    }
+
+    // 2 sponsored shortcuts should be displayed if there are 6 pinned top sites
```

Comment:
This tests the same boundary as the next test. We should keep just 1 (same for manual tests)

Reply:
Done !
Left only the test with 7 pinned top sites.